modules/caddyhttp/reverseproxy/httptransport.go

```diff
@@ -172,38 +173,7 @@ func (h *HTTPTransport) NewTransport(caddyCtx caddy.Context) (*http.Transport, e
 		}
 	}
 
-	// Set up the dialer to pull the correct information from the context
-	dialContext := func(ctx context.Context, network, address string) (net.Conn, error) {
-		// the proper dialing information should be embedded into the request's context
-		if dialInfo, ok := GetDialInfo(ctx); ok {
-			network = dialInfo.Network
-			address = dialInfo.Address
-		}
-
-		conn, err := dialer.DialContext(ctx, network, address)
-		if err != nil {
-			// identify this error as one that occurred during
-			// dialing, which can be important when trying to
-			// decide whether to retry a request
-			return nil, DialError{err}
-		}
-
-		// if read/write timeouts are configured and this is a TCP connection, enforce the timeouts
-		// by wrapping the connection with our own type
-		if tcpConn, ok := conn.(*net.TCPConn); ok && (h.ReadTimeout > 0 || h.WriteTimeout > 0) {
-			conn = &tcpRWTimeoutConn{
-				TCPConn:      tcpConn,
-				readTimeout:  time.Duration(h.ReadTimeout),
-				writeTimeout: time.Duration(h.WriteTimeout),
-				logger:       caddyCtx.Logger(),
-			}
-		}
-
-		return conn, nil
-	}
-
 	rt := &http.Transport{
```

Comment:
I think we still need to use our `dialer` because we customize it for DNS and such. If we leave it full-default then we lose all that too.

```suggestion
	rt := &http.Transport{
		DialContext: func(ctx context.Context, network, address string) (net.Conn, error) {
			return dialer.DialContext(ctx, network, address)
		}
```

Is this what breaks conn pooling? :thinking:

Reply:
I believe DNS resolve is done lines before (by specifying resolver field of dialer), this part deal with unix dialing.

Reply:
But we don't use the `dialer` after that though. We configure it then never use it. This deleted code is where it was actually used.

Reply:
My bad, dialerContext is your suggested change:sweat_smile:.

Reply:
It can be simplified as dialer.DialContext without creating a new function though.